### ISSUE TITLE
use  species filter to reduce metagenotypes displayed?

### ISSUE THREAD

**Issue by ValWood** [pombase/canto#2411]
It might be useful if the organism filter only displayed the metagenotypes for the selected. species (perhaps with an 'All species' option to see everything.

Any ways to reduce the amount of information presented will be useful.  We often want to locate existing similar genotypes to duplicate them and after a while it becomes a lot to scan through.

![image](https://user-images.githubusercontent.com/7359272/107125548-96449600-68a2-11eb-99fe-5bb437ad21ff.png)

@CuzickA would that be useful?

**Comment by jseager7** [comment]
This sounds like a good idea, especially considering that the Genotype Management page only shows the genotypes for the selected organism.

One other thing we might want to consider is additional filters by the pathogen or host side, in case curators want to see all metagenotypes related to a particular pathogen (on all hosts) or vice versa.

**Comment by jseager7** [comment]
I've applied this change to both servers, so it's available for testing by curators.

Under the new system, all metagenotypes are shown when the page is first loaded, and stay shown until _both_ a pathogen and host are selected. This still gives curators the option to see all metagenotypes at once, as in the old system. Unfortunately there's no way to view all metagenotypes after selecting a pathogen and a host, because the organisms can't be deselected. The current solution is also a bit inconsistent with the Genotype Management page, because that doesn't default to showing all genotypes; it just shows no genotypes until an organism is selected.

We might need further discussion to decide on the following points:

* Do we want all metagenotypes shown when the page is first loaded, or should metagenotypes be hidden until a pathogen and a host are selected? (Note that only a filtered list of metagenotypes would be shown in this case.)

* If all metagenotypes are shown when the page is loaded, should we give the curator a way to later return to this view after selecting organisms? If so, how? There are multiple options:

  * Add a control that toggles the filtering, so that curators can switch between all metagenotypes and the filtered metagenotypes. The toggle won't persist between page loads, so hopefully it will be easy for curators to recover if they accidentally toggle the filter off (originally proposed by Val).

  * Add the ability to cancel the selection of the selected pathogen or selected host (or both at once), possibly by adding buttons next to the selector. (This could be awkward since the organism selector code is shared between the Genotype Management page and the Metagenotype Management page, but deselection wouldn't be useful for Genotype Management.)
  
* Should we allow filtering by the pathogen or the host individually (e.g. filter all metagenotypes containing _F. graminearum_, regardless of host)? If so, how should this be controlled by the user interface? Should the filtering apply only when a curator selects a pathogen selecting a host (or vice versa)? This solution might not be useful if the curator has no easy way to deselect organisms.

@ValWood @CuzickA When you have time to test this, could you let me know what you think about the current implementation and the above suggestions?

**Comment by ValWood** [comment]
I haven't had time to test this (have you @CuzickA ?)

@jseager7  could you  demo on the next call? I'm sure it is better....

**Comment by jseager7** [comment]
> jseager7 could you demo on the next call? I'm sure it is better....

Sure. I've added it to the agenda.

**Comment by CuzickA** [comment]
I've not had time to test either, so a demo would be good thanks.

**Comment by jseager7** [comment]
In the last meeting we decided to keep displaying all metagenotypes when the page first loads, and when no pathogen or host is selected.

@ValWood @CuzickA The extra features you requested are ready to be enabled now. I've added the ability to filter by just the pathogen or just the host, when one is selected but not the other. I've also added the ability to deselect the selected pathogen or host by using buttons next to the selector. The buttons are currently only visible on the Metagenotype Management page, but the solution could easily be reused if we decide to do something similar on the Genotype Management page. Here's how it looks:

![image](https://user-images.githubusercontent.com/37659591/112002540-e789ab80-8b17-11eb-9ee3-92ee1f340c97.png)

If there are no objections, I'll enable these changes.
